plugins/aws/lambda/lambdaUniqueExecutionRole.js

```diff
@@ -0,0 +1,53 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Lambda Unique Execution Role',
+    category: 'Lambda',
+    domain: 'Serverless',
+    description: 'Ensure that AWS Lambda functions do not share the same execution role',
+    more_info: 'An execution role grants required permission to Lambda function to access AWS services and resources. It is recommended to associate the unique IAM role for each Lambda function to follow the principle of least privilege access.',
+    link: 'https://docs.aws.amazon.com/lambda/latest/dg/lambda-intro-execution-role.html',
+    recommended_action: 'Modify Lambda function and add new execution role.',
+    apis: ['Lambda:listFunctions'],
+  
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        async.each(regions.lambda, function(region, rcb){
+            var listFunctions = helpers.addSource(cache, source,
+                ['lambda', 'listFunctions', region]);
+
+            if (!listFunctions) return rcb();
+
+            if (listFunctions.err || !listFunctions.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query for Lambda functions: ${helpers.addError(listFunctions)}`, region);
+                return rcb();
+            }
+
+            if (!listFunctions.data.length) {
+                helpers.addResult(results, 0, 'No Lambda functions found', region);
+                return rcb();
+            }
+
+            let executionRoleList = [];
+            for (var lambdaFunc of listFunctions.data) {
+                if (!lambdaFunc.FunctionArn) continue;
+                
+                if (executionRoleList.includes(lambdaFunc.Role)){
+                    helpers.addResult(results, 2, 'Lambda function does not have unique execution role', region, lambdaFunc.FunctionArn);
+                } else {
+                    executionRoleList.push(lambdaFunc.Role);
+                    helpers.addResult(results, 0, 'Lambda function have unique execution role', region, lambdaFunc.FunctionArn);
```

Comment:
isnt that conflicting, if two lambda functions are sharing same execution role, both of lambda's should be producing false result?